go/ct/gen/transient.go

```diff
@@ -0,0 +1,135 @@
+//
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the GNU Lesser General Public Licence v3
+//
+
+package gen
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"golang.org/x/exp/maps"
+	"pgregory.net/rand"
+
+	"github.com/Fantom-foundation/Tosca/go/ct/common"
+	"github.com/Fantom-foundation/Tosca/go/ct/st"
+)
+
+type TransientGenerator struct {
+	set           map[Variable]bool
+	unsatisfiable bool
+}
+
+func NewTransientGenerator() *TransientGenerator {
+	return &TransientGenerator{set: map[Variable]bool{}}
+}
+
+func (t *TransientGenerator) BindSet(key Variable) {
+	if slices.Contains(maps.Keys(t.set), key) {
+		if !t.set[key] {
+			t.unsatisfiable = true
+		}
+	} else {
+		t.set[key] = true
+	}
+}
+
+func (t *TransientGenerator) BindNotSet(key Variable) {
+	if slices.Contains(maps.Keys(t.set), key) {
+		if t.set[key] {
+			t.unsatisfiable = true
+		}
+	} else {
+		t.set[key] = false
+	}
+}
+
+func (t *TransientGenerator) Generate(assignment Assignment, rnd *rand.Rand) (*st.Transient, error) {
+	if t.unsatisfiable {
+		return nil, fmt.Errorf("%w, conflicting set/unset constraints", ErrUnsatisfiable)
```

Comment:
there is already an existing `Err.Unsatisfiable`, maybe use that one ?

Reply:
It is used, it is just wrapped with more information.